client/ayon_core/hosts/houdini/api/hda_utils.py

```diff
@@ -412,3 +412,79 @@ def keep_background_images_linked(node, old_name):
 
     if changes:
         set_background_images(parent, images)
+
+
+def get_products_menu_items(node):
+    """get products menu items
+
+    It gets a list of available products of the specified product types
+      within the specified folder path with in the specified project.
+
+    Users can specify those in the HDA parameters.
+
+    Returns:
+        List[str]: Product options for Products menu.
+    """
+    project_name = node.evalParm("project_name")
+    folder_path = node.evalParm("folder_path")
+    my_product_type = node.evalParm("representation_name")
+
+    id_only = ["id"]
+    folder_entity = ayon_api.get_folder_by_path(project_name,
+                                                folder_path,
+                                                fields=id_only)
+    my_folder_id = folder_entity["id"]
+
+    products = ayon_api.get_products(
+        project_name,
+        folder_ids = [my_folder_id],
+        product_types = [my_product_type]
+
+    )
```

Comment:
```suggestion
    folder_entity = ayon_api.get_folder_by_path(project_name,
                                                folder_path,
                                                fields={"id"})
    if not folder_entity:
        return []

    products = ayon_api.get_products(
        project_name,
        folder_ids=[folder_entity["id"]],
        product_types=[my_product_type]

    )
```

Reply:
I think the product types filtering is incorrect maybe?

Reply:
I can't tell if it's incorrect.